File: app/src/main/java/dev/expositopablo/tonedeath/data/db/DataManager.java
package dev.expositopablo.tonedeath.data.db;

import android.arch.lifecycle.LiveData;
import android.support.annotation.NonNull;

import java.util.List;

import javax.inject.Singleton;

import dev.expositopablo.tonedeath.data.commons.Pinyin;

@Singleton
public interface DataManager {
    void saveScore(int score);
    int getScore();

    LiveData<List<Pinyin>> getAllPinyin();
    LiveData<List<String>> getAllDistinctInitial();
    LiveData<List<String>> getAllDistinctFinal();

    LiveData<List<String>> getMainList(Boolean isInitialFirst);
    LiveData<List<Pinyin>> getDetailList(@NonNull String item, Boolean isInitialFirst);
    Pinyin getRandomPinyin();

    LiveData<List<Pinyin>> getAllPinyinByInitial(@NonNull String pInitial);
    LiveData<List<Pinyin>> getAllPinyinByFinal(@NonNull String pFinal);
}

File: app/src/main/java/dev/expositopablo/tonedeath/data/db/DataManager.kt
package dev.expositopablo.tonedeath.data.db

import dev.expositopablo.tonedeath.data.commons.Pinyin
import io.reactivex.Single

interface DataManager {
    val score: Int

    val allPinyin: Single<List<Pinyin>>
    val allDistinctInitial: Single<List<String>>
    val allDistinctFinal: Single<List<String>>
    val rowCount: Single<Int>
    val randomPinyin: Single<Pinyin>

    fun saveScore(score: Int)

    fun getMainList(isInitialFirst: Boolean): Single<List<String>>
    fun getDetailList(item: String, isInitialFirst: Boolean): Single<List<Pinyin>>

    fun getAllPinyinByInitial(pInitial: String): Single<List<Pinyin>>
    fun getAllPinyinByFinal(pFinal: String): Single<List<Pinyin>>
}
